### ISSUE TITLE
Better handling of unknown AMs

### ISSUE THREAD

**Issue by ahelsing** [GENI-NSF/geni-portal#1346]
If a user submits a request RSpec using an AM URN that is unknown, what should the portal do?

If a user gets a manifest from an AM, one of whose components mentions a component_manager that is unknown, what should the portal do?

Currently the portal displays the URN and component and may be confusing.

 _Imported from trac ticket [#1346](http://trac.gpolab.bbn.com/proto-ch/ticket/1346),  created by ahelsing on 03-17-2015 at 08:25, last modified: 04-01-2015 at 09:57_

**Comment by ahelsing** [comment]
- The portal should not allow you to hit 'Reserve Resources' if the RSpec uses unknown AMs. This may already be true. The portal could tell the experimenter which component uses an unknown AM.
 - When handing off a command to am_client, there is a log message that complains if an AM is unknown, but that message may be failing to log the unknown AM URN. Log more info.
 - What about a manifest RSpec that mentions an unknown AM? Currently the portal lets you try to renew or get status on these components, which then fails. At the very least, the portal should pop up a warning to the experimenter to indicate that it cannot operate on this unknown AM.
 - Do we want to support some way for experimenters to tell the portal about this AM, for purposes of this slice?

_Trac comment by ahelsing on 03-17-2015 at 08:28_

**Comment by ahelsing** [comment]
Some sample log messages we get now:
```
[Tue Mar 31 21:07:55 2015] [error] [client 71.192.37.188] No AM name: Unknown AM URN urn:publicid:IDN+exogeni.net:pscvmsite+authority+am, referer: https://portal.geni.net/secure/sliceresource.php?invocation_user=mzink&invocation_id=ngmiKW&slice_id=4d03a35c-d63e-4990-8607-6ad0c93789ff
[Tue Mar 31 21:25:46 2015] [error] [client 68.42.10.226] Bound RSpec uses unknown AM utc-ig!, referer: https://portal.geni.net/secure/slice-add-resources-jacks.php?slice_id=c24f7234-51c5-4fc0-8a67-514fba3afbcf
[Tue Mar 31 21:25:47 2015] [error] [client 68.42.10.226] am_client cannot do createsliver without an AM URL, referer: https://portal.geni.net/secure/slice-add-resources-jacks.php?slice_id=c24f7234-51c5-4fc0-8a67-514fba3afbcf
...
[Wed Apr 01 08:11:53 2015] [error] [client 130.127.39.240] am_client cannot invoke Omni without an AM URL, referer: https://portal.geni.net/secure/slice.php?slice_id=311125d2-f661-46bf-9200-47393f81d94a
[Wed Apr 01 08:11:53 2015] [error] [client 130.127.39.240] Error result from listresources at  on urn:publicid:IDN+ch.geni.net:gec-demo+slice+gec-prod2: Missing AM URL, referer: https://portal.geni.net/secure/slice.php?slice_id=311125d2-f661-46bf-9200-47393f81d94a
[Wed Apr 01 08:12:52 2015] [error] [client 130.127.39.240] listresources.php: invalid am_id undefined from REQUEST, referer: https://portal.geni.net/secure/slice.php?slice_id=311125d2-f661-46bf-9200-47393f81d94a
[Wed Apr 01 08:12:53 2015] [error] [client 130.127.39.240] PHP Notice:  Undefined variable: ams in /var/www/secure/aggregates.php on line 58, referer: https://portal.geni.net/secure/listresources.php?slice_id=311125d2-f661-46bf-9200-47393f81d94a&am_id[]=undefined
[Wed Apr 01 08:12:53 2015] [error] [client 130.127.39.240] : invalid am_id undefined from REQUEST, referer: https://portal.geni.net/secure/listresources.php?slice_id=311125d2-f661-46bf-9200-47393f81d94a&am_id[]=undefined
```

_Trac comment by ahelsing on 04-01-2015 at 09:57_

**Comment by MarshallBrinn** [comment]
Added a partial (perhaps adequate) fix to #1346. Made rspecuploadparser.php check for unknown aggregates and return unbound or partially bound if an unknown aggregate is bound. As a result, the reserve resources button is disabled until user bounds to known aggregate.

**Comment by MarshallBrinn** [comment]
Closing with fix from 7/1 merge and CHANGES added 7/8.
